benchmark/cuchembm/metrics/embedding.py

```diff
@@ -114,17 +114,19 @@ def encode(self, smiles, zero_padded_vals, average_tokens, max_seq_len=None):
         """Encode a single SMILES to embedding from model"""
         embedding, dim = self._find_embedding(smiles)
         embedding = xpy.array(embedding).reshape(dim).squeeze()
+        n_dim = embedding.ndim
 
         if zero_padded_vals:
-            if dim == 2:
+            if n_dim == 2:
                 embedding[len(smiles):, :] = 0.0
             else:
                 embedding[len(smiles):] = 0.0
 
-        if average_tokens:
-            embedding = embedding[:len(smiles)].mean(axis=0).squeeze()
-        else:
-            embedding = embedding.flatten()
+        if n_dim == 2:
```

Comment:
The dimensionality then has to be added here. What was happening with CDDD was that it was getting averaged down to a single value.